### PR TITLE
Feat: Disable snapshot logic completely if disabled in the config

### PR DESCRIPTION
**Pull request by hmoog**
# Description

This PR completely disables the creation of the snapshot related managers and classes if the features are disabled in the config.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java [hunk 1]
```diff
@@ -433,7 +433,7 @@ private void cleanupExpiredSolidEntryPoints(Tangle tangle, Map<Hash, Integer> ol
                 try {
                     // only clean up if the corresponding milestone transaction was cleaned up already -> otherwise
                     // let the MilestonePrunerJob do this
-                    if (TransactionViewModel.fromHash(tangle, transactionHash).getType() ==
+                    if (transactionPruner != null && TransactionViewModel.fromHash(tangle, transactionHash).getType() ==
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java]: There is no point doing the null check here
You can do it right at the beginning of the method...

File: src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java [hunk 2]
```diff
@@ -464,7 +464,7 @@ private void cleanupOldData(SnapshotConfig config, TransactionPruner transaction
         int startingIndex = config.getMilestoneStartIndex() + 1;
 
         try {
-            if (targetIndex >= startingIndex) {
+            if (transactionPruner != null && targetIndex >= startingIndex) {
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java]: ditto

File: src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java (hunk 1 above)
Comment (hmoog) [reply src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java]: changed it

File: src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java (hunk 2 above)
Comment (hmoog) [reply src/main/java/com/iota/iri/service/snapshot/impl/SnapshotServiceImpl.java]: changed it

**Pull request merged**